### PR TITLE
Decouple the package update and build

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#336936](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/336936)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6899)

### CODE REVIEW

File: src/docfx/serve/lsp/TextDocumentHandler.cs [hunk 1]
```diff
@@ -16,20 +17,42 @@ namespace Microsoft.Docs.Build
 {
     internal class TextDocumentHandler : ITextDocumentSyncHandler
     {
-        private readonly Channel<FileActionEvent> _channel;
-        private readonly DocumentSelector _documentSelector = new(new DocumentFilter { Pattern = "**/*.{md,yml,json}" });
+        private static readonly string[] s_configFileNames =
```
Comment (yufeih) [inline src/docfx/serve/lsp/TextDocumentHandler.cs:20]: is it possible to not specialize config?

File: src/docfx/serve/lsp/TextDocumentHandler.cs [hunk 2]
```diff
@@ -16,20 +17,42 @@ namespace Microsoft.Docs.Build
...
+        private readonly LanguageServerPackage _package;
+
+        private readonly DocumentSelector _documentSelector = new(
+            new DocumentFilter()
+            {
+                Pattern = "**/*.{md,yml,json}",
+            });
 
-        public TextDocumentHandler(Channel<FileActionEvent> channel)
+        public TextDocumentHandler(SemaphoreSlim buildSemaphore, LanguageServerPackage package)
```
Comment (yufeih) [inline src/docfx/serve/lsp/TextDocumentHandler.cs:41]: Remove FileActionEvent.cs?

Comment (yufeih) [inline src/docfx/serve/lsp/TextDocumentHandler.cs:41]: Encapsulate `SemaphoreSlim` as an implementation detail, prefer a method like `QueueBuild`.

```suggestion
        public TextDocumentHandler(LanguageServerBuilder, LanguageServerPackage package)
```

File: docs/specs/lsp.yml [hunk 3]
```diff
@@ -123,3 +123,63 @@ languageServer:
...
+      }
+  - notification: "textDocument/didOpen"
+    params:
+      {
+        "textDocument": {
+          "uri": "file:///{DOCSET_PATH}/../a.md",
+          "text": "Invalid change"
+        }
+      }
+  - expectNoNotification: true
```
Comment (yufeih) [inline docs/specs/lsp.yml:185]: Does it mean this line will wait for 20 seconds?

File: src/docfx/serve/lsp/TextDocumentHandler.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/serve/lsp/TextDocumentHandler.cs:20]: for what reason?

File: docs/specs/lsp.yml (hunk 3 above)
Comment (928PJY) [reply docs/specs/lsp.yml:185]: yes

**Review by yufeih — APPROVED** [review]

File: docs/specs/lsp.yml [hunk 4]
```diff
@@ -123,3 +123,33 @@ languageServer:
         "uri": "file:///{DOCSET_PATH}/b.md",
         "diagnostics": [{ "code": "file-not-found" }]
       }
+---
+# Open/Change of files out of current repo should not trigger build
+os: windows
```
[the remark this answers is not in the document]
Comment (928PJY) [reply docs/specs/lsp.yml:128]: There is still some issue when the test running on the non-windows agent, I will do more investigation leter

**Comment by OsmondJiang** [comment]
the diff is not expected, @fkpwolf would you please help take a look?

**Pull request merged**